qiskit/circuit/gate.py

```diff
@@ -238,7 +241,9 @@ def broadcast_arguments(self, qargs: list, cargs: list) -> Iterable[tuple[list,
         else:
             raise CircuitError("This gate cannot handle %i arguments" % len(qargs))
 
-    def validate_parameter(self, parameter):
+    def validate_parameter(self, parameter, _force_validation=False):
+        if (isinstance(self, StandardGate) or self._skip_validation) and not _force_validation:
```

Comment:
Is this correct? I implemented `RGate` using the scheme in this PR. A test binds a symbolic parameter in an `RGate` in such a way that the numeric result is complex and checks for `CircuitError`. Because of the line above, validation succeeds and the test in the test suite fails.

Reply:
I removed this in: https://github.com/Qiskit/qiskit/pull/12459/commits/39be17b4d5ad2e40a89935d8d960f6d90cbeadd3

Reply:
Since you've removed it, maybe it's a moot point. But I could have been more precise: I find that when you construct a gate like this `RGate(1, 1j)` a `CircuitError` *is* (was) raised as it should be. However, if the complex value arises from calling `assign_parameters`, then the error is *not* raised, when in fact it should be (and was raised before your PR).
The bug only showed up when I moved `RGate` to Rust because `RGate` was the only gate chosen to test this behavior. The test that failed is the second one here:
https://github.com/Qiskit/qiskit/blob/72ff2cf20e1047236c15f27f08c384c40f4a82f8/test/python/circuit/test_parameters.py#L1628-L1643